src/vuex/modules/keystore.js

```diff
@@ -35,19 +35,28 @@ export default () => {
       return getSeed()
     },
     async getAddressFromSeed(store, { seedPhrase, network }) {
-      const networkObject = await this.getNetworkInfo(network, store)
+      const networkObject = await getNetworkInfo(network, store)
       const wallet = await getWallet(seedPhrase, networkObject)
-      return wallet.cosmosAddress
-    },
-    async getNetworkInfo(networkId, store) {
-      return store.getters.networks.find(({ id }) => id === networkId)
+      switch (networkObject.network_type) {
+        case "cosmos": {
+          return wallet.cosmosAddress
+        }
+        case "polkadot": {
+          return JSON.stringify(wallet.publicKey)
```

Comment:
o.O

Reply:
😅 I don't even understand how it was working before, always returning wallet.cosmosAddress O.o

Reply:
the naming is always `cosmosAddress` even for Polkadot. Needs to be adjusted in `cosmos-keys`. But ths line is probably why it doesn't work

Reply:
`getWallet` should actually return `{ address, publicKey, seed }` I think.
For now reverting to `return wallet.cosmosAddress` should work

Reply:
this is actually what `getAddressFromSeed` returns. Let me revert

Reply:
Let me revert